src/main/java/seedu/address/model/module/Module.java

```diff
@@ -58,18 +58,19 @@ public Module(ModuleCode moduleCode, ModuleName moduleName, Description descript
 
     /**
      * Constructs a {@code Module} with only user input fields.
+     * Only used for Tests
      */
-    // public Module(ModuleCode moduleCode, Year year, Semester semester, Grade grade) {
-    //     requireAllNonNull(moduleCode, year, semester, grade);
-    //     this.moduleCode = moduleCode;
-    //     this.yearTaken = year;
-    //     this.semesterTaken = semester;
-    //     this.grade = grade;
-    //     // Temporary until we get back-end setup
-    //     this.moduleName = null;
-    //     this.description = null;
-    //     this.modularCredit = null;
-    // }
+     public Module(ModuleCode moduleCode, Year year, Semester semester, Grade grade) {
+         requireAllNonNull(moduleCode, year, semester, grade);
+         this.moduleCode = moduleCode;
+         this.yearTaken = year;
+         this.semesterTaken = semester;
+         this.grade = grade;
+         // Temporary until we get back-end setup
+         this.moduleName = null;
+         this.description = null;
+         this.modularCredit = null;
+     }
```

Comment:
Shouldn't have to introduce constructors for tests. Violate the concept of a Module, which should always have a name, description and credits.